portfolio-project-matching-app/backend/dao.js

```diff
@@ -1,246 +1,351 @@
-import { getDocSnapshotById, addNewDoc, getCollectionSnapshot, getCollectionSnapshotByCriteria } from '../Firebase/clientApp.ts'
-import { User } from '../models/User'
-import { Project } from '../models/Project'
-import { Technology } from '../models/Technology'
+import {
+    // CREATE
+    addNewDoc,
+    addNewDocWithId,
+    // READ
+    getCollectionReference,
+    getCollectionSnapshot,
+    getDocSnapshotById,
+    // UPDATE
+    updateDocument,
+    // DELETE
+    deleteDocById,
+} from '../Firebase/clientApp.ts'
+
+import {
+    // CREATE
+    createNewLike,
+    // READ
+    getAllProjects,
+    getOwnerByUserId,
+    getProjectById,
+    getTechnologiesByProjectId,
+    getUsersByProjectId,
+    // DELETE
+    deleteLike,
+} from '../backend/daoProject'
+
+import {
+    // READ
+    getAllTechnologies,
+    getTechnologyById,
+} from '../backend/daoTechnology'
+
+import {
+    // READ
+    getAllUsers,
+    getProjectsByUserId,
+    getTechnologiesByUserId,
+    getUserById,
+} from '../backend/daoUser'
+import { DocumentReference } from '@firebase/firestore'
 
 /*
-    PROJECT LEVEL QUERIES
+    CREATE
 */
 
-const getAllProjects = async () => {
+const createDoc = async (coll, payload) => {
```

Comment:
This was one of my newer consolidated / modularized queries that got rid of several `createProject` / `createUser` type queries.